### PR TITLE
import: sst_importer support download SST and rewrite into keyspace data.

### PR DESCRIPTION
**Pull request by iosmanthus**
Signed-off-by: iosmanthus <myosmanthustree@gmail.com>

This pull request depends on https://github.com/tikv/tikv/pull/14027

### What is changed and how it works?

This pull request enables sst_importer to download the SST in keyspace mode, which treats the range info in SST meta and the rewrite rules in plain bytes instead of memory comparable encoded. This might cancel the requirement that old prefixes and new prefixes in rewrite rules should be of the same length.

Issue Number: ref #12999 

What's Changed:

```commit-message
import: sst_importer support download SST and rewrite into keyspace data.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
import: sst_importer support download SST and rewrite into keyspace data.
```

### CODE REVIEW

File: components/sst_importer/src/sst_importer.rs [hunk 1]
```diff
@@ -995,8 +1018,10 @@ impl SstImporter {
         }
 
         // perform iteration and key rewrite.
-        let mut key = keys::data_key(new_prefix);
-        let new_prefix_data_key_len = key.len();
+        let mut data_key = keys::DATA_PREFIX_KEY.to_vec();
+        let data_key_prefix_len = data_key.len();
```
Comment (ystaticy) [inline components/sst_importer/src/sst_importer.rs:1022]: Is it easier to read here if it is written as `let data_key_prefix_len = keys::DATA_PREFIX_KEY.len() ` ?

File: components/sst_importer/src/sst_importer.rs [hunk 2]
```diff
@@ -995,8 +1018,10 @@ impl SstImporter {
...
+        let mut user_key = new_prefix.to_vec();
+        let user_key_prefix_len = user_key.len();
```
Comment (ystaticy) [inline components/sst_importer/src/sst_importer.rs:1024]: Ditto.

File: components/tidb_query_common/src/storage/scanner.rs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.key().
```
Comment (ystaticy) [inline components/tidb_query_common/src/storage/scanner.rs:1]: Does this need to be modified?

**Review by YuJuncen — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Review by pingyu — APPROVED** [review]

**Pull request merged**